packages/cli/src/utils/deploy.ts

```diff
@@ -397,13 +399,27 @@ export async function deploy(
     try {
       const factory = new ethers.ContractFactory(abi, bytecode, signer);
       console.log(chalk.gray(`executing deployment of ${contractName} with nonce ${nonce}`));
-      const deployPromise = factory
-        .deploy({
-          nonce: nonce++,
-          maxPriorityFeePerGas,
-          maxFeePerGas,
-        })
-        .then((c) => (disableTxWait ? c : c.deployed()));
+      let deployPromise;
+
+      if (maxFeePerGas !== undefined && maxPriorityFeePerGas !== undefined) {
+        deployPromise = factory
+          .deploy({
```

Comment:
Since we only set either (`maxFeePerGas` and `maxPriorityFeePerGas`) or (`gasPrice`), with the other one staying `undefined`, we don't need this `if/else` statement but can just pass all three variables to the function.